tutorials/optimization/gpu_optimization.rst

```diff
@@ -0,0 +1,258 @@
+DCCs (such as Blender, Max, etc.) need to keep geometry in CPU memory in order
+for it to be edited, reducing actual performance. Game engines rely on the GPU
+more so they can render many triangles much more efficiently.
+
+On mobile devices, the story is different. PC and Console GPUs are
+brute-force monsters that can pull as much electricity as they need from
+the power grid. Mobile GPUs are limited to a tiny battery, so they need
+to be a lot more power efficient.
+
+To be more efficient, mobile GPUs attempt to avoid *overdraw*. This means, the
+same pixel on the screen being rendered more than once. Imagine a town with
+several buildings, GPUs don't know what is visible and what is hidden until they
+draw it. A house might be drawn and then another house in front of it (rendering
+happened twice for the same pixel!). PC GPUs normally don't care much about this
+and just throw more pixel processors to the hardware to increase performance
+(but this also increases power consumption).
+
+Using more power is not an option on mobile so mobile devices use a technique
+called "Tile Based Rendering" which divides the screen into a grid. Each cell
+keeps the list of triangles drawn to it and sorts them by depth to minimize
+*overdraw*. This technique improves performance and reduces power consumption,
+but takes a toll on vertex performance. As a result, fewer vertices and
+triangles can be processed for drawing.
+
+Additionally, Tile Based Rendering struggles when there are small objects with a
+lot of geometry within a small portion of the screen. This forces mobile GPUs to
+put a lot of strain on a single screen tile which considerably decreases
+performance as all the other cells must wait for it to complete in order to
+display the frame.
+
+In summary, do not worry about vertex count on mobile, but avoid concentration
+of vertices in small parts of the screen. If a character, NPC, vehicle, etc. is
+far away (so it looks tiny), use a smaller level of detail (LOD) model.
+
+Pay attention to the additional vertex processing required when using:
+
+-  Skinning (skeletal animation)
+-  Morphs (shape keys)
+-  Vertex-lit objects (common on mobile)
+
+Pixel / fragment shaders - fill rate
+====================================
+
+In contrast to vertex processing, the costs of fragment shading has increased
+dramatically over the years. Screen resolutions have increased (the area of a 4K
+screen is ``8,294,400`` pixels, versus ``307,200`` for an old ``640x480`` VGA
+screen, that is 27x the area), but also the complexity of fragment shaders has
+exploded. Physically based rendering requires complex calculations for each
+fragment.
+
+You can test whether a project is fill rate limited quite easily. Turn off vsync
+to prevent capping the frames per second, then compare the frames per second
+when running with a large window, to running with a postage stamp sized window.
+Usually you will find the fps increases quite a bit using a small window, which
```

Comment:
postage stamp sized window (you may also benefit from similarly reducing your shadow map size if using shadows)